renku/core/commands/providers/renku.py

```diff
@@ -57,7 +57,13 @@ def find_record(self, uri, client=None):
         failed_urls = []
 
         for kg_url in kg_urls:
-            kg_datasets_url, ssh_url, https_url = self._get_project_urls(kg_url)
+            try:
+                kg_datasets_url, ssh_url, https_url = self._get_project_urls(kg_url)
+            except errors.OperationError as e:
+                # NOTE: Project was likely deleted, but still referenced in the KG
+                if "project not found" not in e.message:
+                    raise
```

Comment:
I'd suggest we raise a specific exception (e.g. `ProjectNotFound`) and catch it here instead of relying on error messages (specially because we might not have a test for this case). Using an existing exception (e.g. `DatasetNotFound`) should be ok too since raise and catch points are close.